app/src/main/java/com/keylesspalace/tusky/components/announcements/AnnouncementAdapter.kt

```diff
@@ -67,6 +71,17 @@ class AnnouncementAdapter(
         fun bind(item: Announcement) {
             text.text = item.content
 
+            val preferences: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(itemView.context)
+            val wellbeingEnabled = preferences.getBoolean(PrefKeys.WELLBEING_HIDE_STATS_POSTS, false)
```

Comment:
I'm afraid, the `bind` method is not the right place to load the settings. It might be called quite often, loading preferences can be expensive (disk read!) and the value of the setting will not change. Better load it outside of the adapter and pass it through the constructor or something like that.

Reply:
Right. I totally overlooked this. Thanks for the hint.